src/oop/object_independence/hard/q2/MainTest.java

```diff
@@ -0,0 +1,299 @@
+                {gender, String.class},
+                {age, int.class}
+        };
+        Object instance = classInstance.createInstance(arguments);
+        String expectedOutput = "This fish is " + colour + ", and is " + gender + ", and is " + age + " years old";
+        Object output = classInstance.callMethod("toString", new String[]{"public"}, instance);
+        String incorrectToString = String.format("Your toString method for the %s class does return the correct String.", objectClassName);
+        _assertEquals(expectedOutput, output, incorrectToString);
+    }
+
+    @ParameterizedTest
+    @MethodSource("setMethodsInputProvider")
+    public void fishMethodsWorkCorrectlyWorkTogether(String colour, String gender, int age, String newColour, String newGender, int newAge) throws Throwable {
+        Object[][] constructorArguments = {
+                {colour, String.class},
+                {gender, String.class},
+                {age, int.class}
+        };
+        Object instance = classInstance.createInstance(constructorArguments);
+        classInstance.callMethod(setStringAttributeName1, new Object[][]{{newColour, String.class}}, instance);
+        classInstance.callMethod(setStringAttributeName2, new Object[][]{{newGender, String.class}}, instance);
+        classInstance.callMethod(setIntAttributeName1, new Object[][]{{newAge, int.class}}, instance);
+        Object output1 = classInstance.callMethod(getStringAttributeName1, instance);
+        Object output2 = classInstance.callMethod(getStringAttributeName2, instance);
+        Object output3 = classInstance.callMethod(getIntAttributeName1, instance);
+        Object toStringOutput = classInstance.callMethod("toString", instance);
+        String expectedToStringOutput = "This fish is " + newColour + ", and is " + newGender + ", and is " + newAge + " years old";
+        String incorrectSetGet =
+                "Your %s method does not return the correct value after updating the %s attribute using the %s method.";
+        String incorrectToString = "Your toString method does not return the correct String after updating the %s, %s, and %s attributes using the %s, %s, and %s methods.";
+        _assertEquals(newColour, output1,
+                String.format(incorrectSetGet, getStringAttributeName1, stringAttributeName1, setStringAttributeName1));
+        _assertEquals(newGender, output2,
+                String.format(incorrectSetGet, getStringAttributeName2, stringAttributeName2, setStringAttributeName2));
+        _assertEquals(newAge, output3,
+                String.format(incorrectSetGet, getIntAttributeName1, intAttributeName1, setIntAttributeName1));
+        _assertEquals(expectedToStringOutput, toStringOutput,
+                String.format(incorrectToString, stringAttributeName1, stringAttributeName2, intAttributeName1,
+                        setStringAttributeName1, setStringAttributeName2, setIntAttributeName1));
+    }
+
+    @Test
```

Comment:
This test should be dependent on others using tags. Please see https://github.com/canvas-gamification/canvas-gamification-junit-tests/pull/694 for an example.